diff --git a/src/views/dialogue/components/DialogueSession.vue b/src/views/dialogue/components/DialogueSession.vue
index 4fd354b..4096860 100644
--- a/src/views/dialogue/components/DialogueSession.vue
+++ b/src/views/dialogue/components/DialogueSession.vue
@@ -30,6 +30,10 @@ import pptxSvg from '@/assets/svgs/pptx.svg';
 import yamlSvg from '@/assets/svgs/yaml.svg';
 import zipSvg from '@/assets/svgs/zip.svg';
 import { ElMessage } from 'element-plus';
+import { useAccountStore } from 'src/store';
+import { nextTick } from 'vue';
+const autoExecuteRef = ref(<boolean | undefined>false);
+const { userinfo } = storeToRefs(useAccountStore());
 
 const typeSvgMap = {
   doc: docSvg,
@@ -86,8 +90,9 @@ const dialogueInput = ref<string>('');
 
 // 对话列表
 const { sendQuestion } = useSessionStore();
-const { conversationList, isAnswerGenerating, dialogueRef } =
-  storeToRefs(useSessionStore());
+const { conversationList, isAnswerGenerating, dialogueRef } = storeToRefs(
+  useSessionStore(),
+);
 const { generateSession } = useHistorySessionStore();
 const { currentSelectedSession } = storeToRefs(useHistorySessionStore());
 /**
@@ -166,7 +171,7 @@ const handleKeydown = (event: KeyboardEvent) => {
  *
  * @param item
  */
-const getItem = <T,>(item: ConversationItem, field: string): T | undefined => {
+const getItem = <T>(item: ConversationItem, field: string): T | undefined => {
   if (field in item) {
     return (item as RobotConversationItem)[field] as T;
   }
@@ -554,6 +559,13 @@ const getProviderLLM = async () => {
   }
 };
 
+const autoExecuteChange = async (value) => {
+  autoExecuteRef.value = value;
+  await nextTick();
+  Object.assign(userinfo.value, { auto_execute: value });
+  api.updateUserInfo({ autoExecute: value });
+};
+
 onMounted(() => {
   // 数据初始化
   AppForm.value = props.createAppForm;
@@ -562,6 +574,19 @@ onMounted(() => {
   getProviderLLM();
 });
 
+watch(
+  [userinfo],
+  () => {
+    if (userinfo.value.auto_execute) {
+      autoExecuteRef.value = userinfo.value.auto_execute;
+    }
+  },
+  {
+    immediate: true,
+    deep: true,
+  },
+);
+
 watch(selectLLM, (newValue) => {
   if (newValue) {
     selectedLLM.value.modalName = newValue.modelName;
@@ -638,7 +663,7 @@ watch(
 watch(
   () => app,
   (val) => {
-    if (app.value) {
+    if (app.value.appId) {
       user_selected_app.value = app.value.appId;
     }
   },
@@ -679,14 +704,16 @@ function downloadFun(url: string) {
       let fileName = 'default-filename';
       // 解析文件名（处理编码及格式）
       if (contentDisposition) {
-        const fileNameMatch = contentDisposition.match(/filename\*?=(?:UTF-8'')?"?([^";]+)"?/i);
+        const fileNameMatch = contentDisposition.match(
+          /filename\*?=(?:UTF-8'')?"?([^";]+)"?/i,
+        );
         if (fileNameMatch && fileNameMatch[1]) {
           fileName = decodeURIComponent(fileNameMatch[1].replace(/"/g, ''));
         }
       }
-      return response.blob().then((blob) => ({ blob,fileName }));
+      return response.blob().then((blob) => ({ blob, fileName }));
     })
-    .then(({ blob,fileName }) => {
+    .then(({ blob, fileName }) => {
       const a = document.createElement('a');
       a.href = URL.createObjectURL(blob);
       a.download = fileName;
@@ -700,7 +727,7 @@ function downloadFun(url: string) {
 }
 const downLoadSourceFile = (file: any) => {
   if (!file) return;
-  const url = `${window.origin}/wtd/api/doc/download?docId=${file.documentId}`;
+  const url = `${window.origin}/witchaind/api/doc/download?docId=${file.documentId}`;
   downloadFun(url);
 };
 
@@ -710,19 +737,19 @@ const formatDate = (timestamp: number) => {
   const month = String(date.getMonth() + 1).padStart(2, '0');
   const day = String(date.getDate()).padStart(2, '0');
   return `${year}-${month}-${day}`;
-}
+};
 /**
-  * @description 处理并过滤文件列表，将文件列表中的字段名统一为指定格式
-  * @param {ConversationItem} ConversationItem - 对话项对象
-  * @param {string} str - 字段名
-  * @returns {Array} 格式化后的文件列表
+ * @description 处理并过滤文件列表，将文件列表中的字段名统一为指定格式
+ * @param {ConversationItem} ConversationItem - 对话项对象
+ * @param {string} str - 字段名
+ * @returns {Array} 格式化后的文件列表
  */
-const getFormatFileList = (ConversationItem,str)=>{
-  let fileList: any= getItem(ConversationItem,str);
+const getFormatFileList = (ConversationItem, str) => {
+  let fileList: any = getItem(ConversationItem, str);
   if (!fileList || fileList?.length === 0) return;
-  let newFileList:any = [];
-   fileList?.forEach((file) => {
-    if(file.associated === 'answer'){
+  let newFileList: any = [];
+  fileList?.forEach((file) => {
+    if (file.associated === 'answer') {
       newFileList.push({
         documentId: file._id,
         documentName: file.name,
@@ -733,11 +760,11 @@ const getFormatFileList = (ConversationItem,str)=>{
         documentOrder: file.order,
         createdAt: file.created_at,
         documentAuthor: file.author,
-      })
+      });
     }
-  })
+  });
   return newFileList;
-}
+};
 </script>
 
 <template>
@@ -825,7 +852,10 @@ const getFormatFileList = (ConversationItem,str)=>{
             {{ $t('feedback.stop') }}
           </div>
         </div>
-        <div class="dialogue-conversation-bottom-selectGroup">
+        <div
+          class="dialogue-conversation-bottom-selectGroup"
+          v-if="!user_selected_app"
+        >
           <div class="modalSelectGroup">
             <el-dropdown
               trigger="click"
@@ -851,7 +881,7 @@ const getFormatFileList = (ConversationItem,str)=>{
                 </el-icon>
               </div>
               <div class="el-dropdown-link" v-else>
-                <span>请选择模型</span>
+                <span>{{ $t('app.modelSelected_input') }}</span>
                 <el-icon
                   :class="{ rotated: isDropdownOpen }"
                   style="margin-left: auto"
@@ -893,7 +923,10 @@ const getFormatFileList = (ConversationItem,str)=>{
               />
             </div>
             <!-- 上传 -->
-            <div class="dialogue-conversation-bottom-sendbox__upload">
+            <div
+              class="dialogue-conversation-bottom-sendbox__upload"
+              v-if="!user_selected_app"
+            >
               <el-tooltip
                 placement="top"
                 :content="$t('upload.upload_tip_text')"
@@ -921,6 +954,38 @@ const getFormatFileList = (ConversationItem,str)=>{
                 </div>
               </el-tooltip>
             </div>
+            <!-- 自动执行 -->
+            <div
+              class="dialogue-conversation-bottom-sendbox__autoExecute"
+              v-if="user_selected_app"
+            >
+              <el-tooltip
+                placement="top"
+                :content="$t('history.auto_execute')"
+                effect="light"
+              >
+                <div class="switch-wrapper">
+                  <span
+                    style="
+                      width: 70px;
+                      display: inline-block;
+                      overflow: hidden;
+                      text-overflow: ellipsis;
+                      white-space: nowrap;
+                      vertical-align: middle;
+                    "
+                  >
+                    {{ $t('history.auto_execute') }}
+                  </span>
+                  <el-switch
+                    v-model="autoExecuteRef"
+                    :active-value="true"
+                    :inactive-value="false"
+                    @change="autoExecuteChange"
+                  ></el-switch>
+                </div>
+              </el-tooltip>
+            </div>
             <!-- 发送问题 -->
             <div class="dialogue-conversation-bottom-sendbox__icon">
               <img
@@ -961,7 +1026,7 @@ const getFormatFileList = (ConversationItem,str)=>{
     <div v-if="showFileSource" class="dialogue-rightContainer__file-source">
       <div class="file-source__header">
         <div class="file-source__title">
-          <span>引用来源</span>
+          <span>{{ $t('upload.reference_source') }}</span>
           <span>{{ curFileList.length || 0 }}</span>
         </div>
         <el-icon @click="closeShowFileSource">
@@ -970,23 +1035,31 @@ const getFormatFileList = (ConversationItem,str)=>{
       </div>
       <div class="file-source__content-container">
         <div class="file-source__content">
-          <div v-for="(file,index) in curFileList" :key="index" class="file-source__item">
+          <div
+            v-for="(file, index) in curFileList"
+            :key="index"
+            class="file-source__item"
+          >
             <div class="file-source__item-header">
               <div class="file-source__item-header-left">
-                <span class="file-source__order">{{ file?.documentorder ?? index+1 }}</span>
+                <span class="file-source__order">
+                  {{ file?.documentorder ?? index + 1 }}
+                </span>
                 <div class="file-source__type-icon">
-                  <img :src="typeSvgMap[file?.documentType]" alt="">
+                  <img :src="typeSvgMap[file?.documentType]" alt="" />
                 </div>
                 <span class="file-source__name">{{ file?.documentName }}</span>
               </div>
-              <el-tooltip
-                content="下载"
-                placement="top"
-              >
-                <span class="file-source__download" @click="downLoadSourceFile(file)"></span>
+              <el-tooltip content="下载" placement="top">
+                <span
+                  class="file-source__download"
+                  @click="downLoadSourceFile(file)"
+                ></span>
               </el-tooltip>
             </div>
-            <span class="file-source__desc">{{ file?.documentAbstract ?? '' }}</span>
+            <span class="file-source__desc">
+              {{ file?.documentAbstract ?? '' }}
+            </span>
             <div class="file-source__footer">
               <span>@{{ file?.documentAuthor }}</span>
               <span>{{ formatDate(file?.createdAt) }}</span>
@@ -1200,19 +1273,9 @@ button[disabled]:hover {
           color: var(--o-text-color-primary);
           font-size: 16px;
           background-color: var(--o-bg-color-base);
-          font-family:
-            HarmonyOS_Sans_SC_Regular,
-            system-ui,
-            -apple-system,
-            BlinkMacSystemFont,
-            'Segoe UI',
-            Roboto,
-            Oxygen,
-            Ubuntu,
-            Cantarell,
-            'Open Sans',
-            'Helvetica Neue',
-            sans-serif;
+          font-family: HarmonyOS_Sans_SC_Regular, system-ui, -apple-system,
+            BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell,
+            'Open Sans', 'Helvetica Neue', sans-serif;
 
           &:focus {
             outline: none;
@@ -1258,6 +1321,25 @@ button[disabled]:hover {
         }
       }
 
+      &__autoExecute {
+        position: absolute;
+        cursor: pointer;
+        width: 120px;
+        height: 32px;
+        box-sizing: border-box;
+        border: 1px solid rgba(223, 229, 239);
+        border-radius: 8px;
+        background: rgb(253, 254, 255);
+
+        .switch-wrapper {
+          position: absolute;
+          width: 120px;
+          height: 32px;
+          line-height: 30px;
+          text-align: center;
+        }
+      }
+
       &__icon {
         text-align: right;
         & img {
@@ -1269,7 +1351,7 @@ button[disabled]:hover {
   }
 }
 
-.dialogue-rightContainer__file-source{
+.dialogue-rightContainer__file-source {
   width: 342px;
   background-color: var(--o-bg-color-base);
   margin-top: 16px;
@@ -1279,36 +1361,36 @@ button[disabled]:hover {
   padding: 16px;
   display: flex;
   flex-direction: column;
-  .file-source__header{
+  .file-source__header {
     display: flex;
     justify-content: space-between;
     align-items: center;
-    .file-source__title{
+    .file-source__title {
       display: flex;
       gap: 8px;
-      span{
+      span {
         font-size: 16px;
         color: var(--o-text-color-primary);
         font-weight: 700;
         line-height: 24px;
       }
     }
-    .el-icon{
+    .el-icon {
       height: 16px;
       cursor: pointer;
     }
   }
-  .file-source__content-container{
+  .file-source__content-container {
     max-height: 100vh;
     overflow-y: auto;
     margin-right: -4px;
     margin-top: 8px;
-    .file-source__content{
+    .file-source__content {
       display: flex;
       flex-direction: column;
       gap: 8px;
       padding: 2px;
-      .file-source__item{
+      .file-source__item {
         background-color: var(--o-bg-color-light);
         width: 310px;
         border-radius: 8px;
@@ -1316,14 +1398,14 @@ button[disabled]:hover {
         &:hover {
           outline: 2px solid #7aa5ff;
         }
-        .file-source__item-header{
+        .file-source__item-header {
           display: flex;
           justify-content: space-between;
           align-items: flex-start;
           gap: 4px;
-          .file-source__item-header-left{
+          .file-source__item-header-left {
             max-width: 258px;
-            .file-source__order{
+            .file-source__order {
               display: inline-block;
               font-size: 12px;
               color: var(--o-text-color-primary);
@@ -1335,41 +1417,45 @@ button[disabled]:hover {
               border-radius: 50%;
               text-align: center;
             }
-            .file-source__type-icon{
+            .file-source__type-icon {
               display: inline-flex;
               align-items: center;
               vertical-align: bottom;
               margin-right: 2px;
               height: 22px;
-              img{
+              img {
                 width: 16px;
                 height: 16px;
               }
             }
-            .file-source__name{
+            .file-source__name {
               font-size: 14px;
               color: var(--o-text-color-primary);
               font-weight: 700;
               line-height: 22px;
             }
           }
-          .file-source__download{
+          .file-source__download {
             cursor: pointer !important;
             display: inline-flex;
             align-items: center;
             vertical-align: bottom;
             height: 22px;
             width: 16px;
-            background: url('@/assets/svgs/file_download.svg') no-repeat center center;
-            &:hover{
-              background: url('@/assets/svgs/file_download_hover.svg') no-repeat center center;
+            background: url('@/assets/svgs/file_download.svg') no-repeat center
+              center;
+            &:hover {
+              background: url('@/assets/svgs/file_download_hover.svg') no-repeat
+                center center;
             }
-            &:active,&:focus{
-              background: url('@/assets/svgs/file_download_active.svg') no-repeat center center;
+            &:active,
+            &:focus {
+              background: url('@/assets/svgs/file_download_active.svg')
+                no-repeat center center;
             }
           }
         }
-        .file-source__desc{
+        .file-source__desc {
           font-size: 12px;
           color: var(--o-text-color-secondary);
           line-height: 16px;
@@ -1382,7 +1468,7 @@ button[disabled]:hover {
           text-overflow: ellipsis;
           word-break: break-all;
         }
-        .file-source__footer{
+        .file-source__footer {
           font-size: 12px;
           display: flex;
           justify-content: space-between;
