diff --git a/src/views/createapp/components/workFlowConfig/workFlowDialog.vue b/src/views/createapp/components/workFlowConfig/workFlowDialog.vue
index 25ed229..b66a5ff 100644
--- a/src/views/createapp/components/workFlowConfig/workFlowDialog.vue
+++ b/src/views/createapp/components/workFlowConfig/workFlowDialog.vue
@@ -126,7 +126,7 @@ const handleSubmit = (formEl: FormInstance | undefined) => {
                   apiId: 'startId',
                   nodeId: 'Empty',
                   serviceId: 'start',
-                  name: '开始',
+                  name: i18n.global.t('main.start'),
                   callId: 'start',
                   description: 'startNode',
                   editable: false,
@@ -141,7 +141,7 @@ const handleSubmit = (formEl: FormInstance | undefined) => {
                   apiId: 'endId',
                   nodeId: 'Empty',
                   serviceId: 'end',
-                  name: '结束',
+                  name: i18n.global.t('main.end') ,
                   callId: 'end',
                   description: 'endNode',
                   editable: false,
